lib/Analysis/Membar.cpp

```diff
@@ -73,8 +73,10 @@ void MembarAnalysis::visitTerminator(Operation *op,
     }
     return;
   }
-  // Otherwise, it could be a return op
-  assert(isa<func::ReturnOp>(op) && "Unknown terminator");
+  if (isa<triton::ReduceReturnOp>(op) || isa<func::ReturnOp>(op)) {
```

Comment:
It's better to use assert here or reachable code I think. as emitOpError doesn't terminate the execution.

Reply:
Changed to `llvm_unreachable`.